src/argilla/training/openai.py

```diff
@@ -0,0 +1,223 @@
+        return "\nhttps://platform.openai.com/docs/api-reference/fine-tune\n" + "\n".join(formatted_string)
+
+    def upload_dataset_to_openai(self, dataset, file_name):
+        import json
+
+        import openai
+
+        for rec in dataset:
+            del rec["id"]
+        with open(file_name, "w") as f:
+            for item in dataset:
+                f.write(json.dumps(item) + "\n")
+        upload_response = openai.File.create(file=open(file_name, "rb"), purpose="fine-tune")
+        self._logger.info(upload_response)
+        file_id = upload_response.id
+        return file_id
+
+    def train(self, output_dir: str = None):
+        """
+        We create a openai.FineTune object from a pretrained model, and send data to finetune it.
+        """
+
+        import openai
+
+        if output_dir is not None:
+            self.model_kwargs["suffix"] = output_dir
+
+        if self._train_dataset is not None and self.model_kwargs["training_file"] is None:
+            self.model_kwargs["training_file"] = self.upload_dataset_to_openai(self._train_dataset, "data_train.jsonl")
+        if self._eval_dataset is not None and self.model_kwargs["validation_file"] is None:
+            self.model_kwargs["validation_file"] = self.upload_dataset_to_openai(self._eval_dataset, "data_test.jsonl")
+
+        self.update_config()
+
+        response = openai.FineTune.create(
+            **self.model_kwargs,
+        )
+        self._logger.info(response)
+        self.finetune_id = response.id
+
+    def init_model(self):
+        import openai
+
+        if self.finetune_id is not None:
+            response = openai.FineTune.retrieve(self.finetune_id)
+            potential_model = response.fine_tuned_model
+            if potential_model is None:
+                self._logger.warning("Fine-tuning is still in progress.")
+            else:
+                self._model = potential_model
+        else:
+            response = openai.Model.retrieve(self._model)
+            self._model = response.id
+            self._logger.info(response)
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True):
+        """
+        The function takes in a list of strings and returns a list of predictions
+
+        Args:
+          text (Union[List[str], str]): The text to be classified.
+          as_argilla_records (bool): If True, the prediction will be returned as an Argilla record. If
+        False, the prediction will be returned as a string. Defaults to True
+
+        Returns:
+          A list of predictions
+        """
+
+        import openai
+
+        responses = []
+        was_string = False
+
+        if isinstance(text, str):
+            text = [text]
+            was_string = True
+
+        if isinstance(self._settings, TextClassificationSettings):
+            kwargs = {"logprobs": len(self._settings.label_schema), "max_tokens": 1}
+
+        for entry in text:
+            prompt = entry + self._separator
```

Comment:
Here maybe we can just `strip()` the str before adding the separator in case there's any line break in the string.

```suggestion
            prompt = f"{entry.strip()}{self._separator}"
```

Reply:
Strip does not remove line breaks unless they are at the start/end of the string. I also think the f-string formatting is a bit overkill if we're only concatenating strings. @davidberenstein1957

Reply:
Yes, but since the separator is `\n\n###\n\n` we should be removing at least the line breaks at the end, since the tokenizer counts the line breaks as tokens, and I feel like removing the extra line breaks is more efficient here. Both at the start and at the end, since the rest of the line breaks may be on purpose.

Reply:
Also, regarding the f-string I'd say that the performance compared to string concatenation is pretty similar indeed. But we can rely on string concatenation for now!